config/peerpods/podvm/aws-podvm-image-handler.sh

```
@@ -215,13 +246,26 @@ function prepare_for_prebuilt_artifact() {
+        echo "Using IRSA credentials for image creation with role: ${AWS_ROLE_ARN}"
+        /scripts/ami-helper.sh -s -b ${BUCKET_NAME} || error_exit "Failed to create bucket using the helper script"
+    elif is_cco_flow_secret; then
+        echo "CCO flow detected, requesting credential extension"
+        /scripts/ami-helper.sh -i -b ${BUCKET_NAME} || error_exit "Failed to extend credentials using the helper script"
+        export AWS_ACCESS_KEY_ID=$(kubectl get secret ${extended_secret_name} -n openshift-sandboxed-containers-operator -o jsonpath='{.data.aws_access_key_id}' | base64 -d)
+        export AWS_SECRET_ACCESS_KEY=$(kubectl get secret ${extended_secret_name} -n openshift-sandboxed-containers-operator -o jsonpath='{.data.aws_secret_access_key}' | base64 -d)
+    else
+        echo "Using manual credentials (expecting s3 ${BUCKET_NAME} bucket to exist and vmimport service role to be set)"
+
+        # Validate bucket exists
+        aws s3api head-bucket --bucket ${BUCKET_NAME} --region ${AWS_REGION} 2>/dev/null || \
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)